system/lib/libc/emscripten_internal.h

```diff
@@ -132,6 +133,10 @@ void _emscripten_fetch_free(unsigned int);
 
 EMSCRIPTEN_RESULT _emscripten_set_offscreencanvas_size(const char *target, int width, int height);
 
+// Internal implementation function in JavaScript side that emscripten_create_wasm_worker() calls to
+// to perform the wasm worker creation.
+emscripten_wasm_worker_t _emscripten_create_wasm_worker(void *stackLowestAddress, uint32_t stackSize);
+
```

Comment:
Why put this function here? This function is here right next to offscreen canvas code, which does not relate to wasm workers? A general "all internal code" kind of header does not seem encapsulating, but creates false code include dependencies?

Reply:
This header is a catch-all for **all** internal emscripten JS APIs that don't have a natural place in any other internal header (e.g. `threading_internal.h`).

Instead of this giant header we could split it into `wasm_worker_internal.h` and `proxying_internal.h`, but since its internal-only,  I doesn't seem very important, and we can change it whenever we want.

Reply:
Gotcha. I do think having the declaration in `library_wasm_worker.c` would be better code style. It explicitly says that "only this file is going to (intended to) call this internal function, and this is encapsulated to be Wasm Workers specific code".

By reading a declaration of `_emscripten_create_wasm_worker` in `emscripten_internal.h`, it does not give any static hint towards what might be consuming that function, and/or the intended code structure. This migration feels like is losing something and reduces cohesion of the code a bit.

Not a big deal either way though.

Reply:
The way the signature generations works is that we want to be able to include references to all JS API symbols from a single auto-generated `.c` file.   This `.c` file needs to have the correct signatures for all the JS API symbol (both internal and external), so it needs to see the declarations of the function somehow. 

I could duplicate the declarations of these internal functions, but that would risk the signatures becoming out-of-sync.

Reply:
Also see the comment at the top of this header: https://github.com/emscripten-core/emscripten/blob/9dff99d6a303c7987fe195a53c883b1c1b390b70/system/lib/libc/emscripten_internal.h#L7-L11

I'm open to ideas on how to improve this, but for now, especially since this header is not available externally, I think it works well enough.